java/org/apache/coyote/http11/AbstractHttp11Protocol.java

```diff
@@ -244,6 +244,34 @@ public void setMaxSavePostSize(int maxSavePostSize) {
     public void setMaxHttpHeaderSize(int valueI) { maxHttpHeaderSize = valueI; }
 
 
+    /**
+     * Maximum size of the HTTP request message header.
+     */
+    private int maxHttpRequestHeaderSize = -1;
```

Comment:
After the change, `maxHttpHeaderSize` becomes default maximum size of the HTTP message header for both request and response.

If we no longer have consumer of `getMaxHttpHeaderSize()`, then I'd suggest to remove `maxHttpHeaderSize`, and change `setMaxHttpHeaderSize()` like this 

```java
@Deprecated
public void setMaxHttpHeaderSize(int valueI) {
    maxHttpRequestHeaderSize = valueI;
    maxHttpResponseHeaderSize = valueI;
}
```

Reply:
I'm worried about the compatibility risk of removing getMaxHttpHeaderSize(), because this is a public API after all, I wonder if this change can be done in 10.1.x. of which I agree with the annotation Deprecated.

Reply:
Keep `getMaxHttpHeaderSize()` only preserves API level compatibility, the semantic of `getMaxHttpHeaderSize()` has change once we introduce dedicated header settings for request and response.

To illustrate the semantic change:
```java
setMaxHttpHeaderSize(4096);
setMaxHttpRequestHeaderSize(8192);
setMaxHttpResponseHeaderSize(16384);
```
Then `getMaxHttpHeaderSize()` make no sense except the only meaning "default maxHttpHeaderSize"

Maybe it is  deprecate `getMaxHttpHeaderSize()` and return a size that is not too bad:
```java
public int getMaxHttpHeaderSize() {
    return Math.min(maxHttpRequestHeaderSize, maxHttpResponseHeaderSize)
}
```

I still insist that semantic level compatibility is more important than API level compatibility. For `setMaxHttpHeaderSize()` it originally means set maximum size of the HTTP message header for both request and response.